### PR TITLE
Sol 1

### PR DESCRIPTION
**Pull request by Unlie9**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,41 @@
...
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(herbivore: Herbivore) -> None:
+        if not isinstance(herbivore, Herbivore):
```
Comment (DevSpaciX) [inline app/main.py:32]: You can add `or` param and check if the `herbivore` is hidden cuz we do not need to change health in that case , yes ?

File: app/main.py [hunk 2]
```diff
@@ -1 +1,41 @@
...
+            return
+
+        if herbivore.hidden is False:
+            herbivore.health -= 50
+            if herbivore.health <= 0:
+                herbivore.die()
+
+        if herbivore.hidden is True:
```
Comment (DevSpaciX) [inline app/main.py:40]: So we just passing the default param ?

File: app/main.py [hunk 3]
```diff
@@ -1 +1,41 @@
...
+class Carnivore(Animal):
+    @staticmethod
+    def bite(herbivore: Herbivore) -> None:
+        if not isinstance(herbivore, Herbivore):
+            return
+
+        if herbivore.hidden is False:
+            herbivore.health -= 50
+            if herbivore.health <= 0:
+                herbivore.die()
```
Comment (DevSpaciX) [inline app/main.py:38]: Not bad idea , but could we just call remove method here ?

**Review by DevSpaciX — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (DevSpaciX) [reply app/main.py:28]: @Unlie9 I think you forget about this

File: app/main.py (hunk 2 above)
Comment (DevSpaciX) [reply app/main.py]: I was about to catch this case and do nothing cuz this is default params : )

File: app/main.py [hunk 4]
```diff
@@ -1 +1,39 @@
...
+            return
+
+        if herbivore.hidden:
+            herbivore.health = 100
+        else:
+            herbivore.health -= 50
+            if herbivore.health <= 0:
+                Animal.alive.remove(herbivore)
+
+        if herbivore.hidden:
```
Comment (DevSpaciX) [inline app/main.py:38]: I was about to catch this case and do nothing cuz this is default params : )

**Review by DevSpaciX — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 4 above)
Comment (Unlie9) [reply app/main.py]: i got you

File: app/main.py [hunk 5]
```diff
@@ -1 +1,36 @@
...
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(herbivore: Herbivore) -> None:
+        if not isinstance(herbivore, Herbivore):
```
Comment (DevSpaciX) [inline app/main.py:28]: ` if not isinstance(herbivore, Herbivore) or herbivore.hidden:`
    `return`

File: app/main.py [hunk 6]
```diff
@@ -1 +1,36 @@
...
+            return
+
+        if herbivore.hidden:
+            herbivore.health = 100
```
Comment (DevSpaciX) [inline app/main.py:32]: so now you can delete this code

**Review by DevSpaciX — APPROVED** [review]

**Pull request closed without merging**